addons/mrp/models/stock_move.py

```diff
@@ -69,6 +69,14 @@ class StockMove(models.Model):
     needs_lots = fields.Boolean('Tracking', compute='_compute_needs_lots')
     order_finished_lot_ids = fields.Many2many('stock.production.lot', compute='_compute_order_finished_lot_ids')
 
+    @api.model
+    def create(self, vals):
+        # quantity_done is only explicitly set when the user creates the move and specifies it manually
+        quantity_done = vals.get('quantity_done')
+        if vals.get('raw_material_production_id') and quantity_done is not None and quantity_done <= 0:
```

Comment:
< since it's possible to consume 0 qty since https://github.com/odoo/odoo/commit/4cb2e626a9ea30ea05e2844a29ca8eec6f2bc567